nl2sql/sql_chain.py

```diff
@@ -15,10 +15,9 @@ def sql_chain(input: str):
     result = db_chain(input)
 
     sql_query = result["intermediate_steps"][1]
-    sql_results = ast.literal_eval(result["intermediate_steps"][3])
-    answer = result["intermediate_steps"][5]
+    sql_results = execute_query_and_return_df(sql_query, db_path)
```

Comment:
just call the method `execute_query`. it is not common to add the types to the method name.

Reply:
You could extract this entire code to a different function, so it is clear that it's only purpose is to translate a natural language into an sql query.
```
 db = SQLDatabase.from_uri(f"sqlite:///{db_path}", sample_rows_in_table_info=3)
    db_chain = SQLDatabaseChain.from_llm(
        _create_llm(), db, verbose=True, return_intermediate_steps=True
    )
    result = db_chain(input)

    sql_query = result["intermediate_steps"][1]
    ```